avm/res/service-fabric/cluster/main.bicep

```
@@ -29,17 +29,17 @@ param maxUnusedVersionsToKeep int = 3
 @description('Optional. The settings to enable AAD authentication on the cluster.')
 param azureActiveDirectory object = {}
 
-@description('Optional. Describes the certificate details like thumbprint of the primary certificate, thumbprint of the secondary certificate and the local certificate store location.')
-param certificate object = {}
+@description('Conditional. The certificate to use for securing the cluster. The certificate provided will be used for node to node security within the cluster, SSL certificate for cluster management endpoint and default admin client. Required if the certificateCommonNames parameter is not used.')
+param certificate certificateType
 
-@description('Optional. Describes a list of server certificates referenced by common name that are used to secure the cluster.')
-param certificateCommonNames object = {}
+@description('Conditional. Describes a list of server certificates referenced by common name that are used to secure the cluster. Required if the certificate parameter is not used.')
+param certificateCommonNames certificateCommonNamesType
```

For consistency with the others I'd suggest to keep the UDT name in singular form, like:
```suggestion
param certificateCommonNames certificateCommonNameType
```